daemon/storage.go

```diff
@@ -400,6 +404,70 @@ func (o *OverlayFsStorage) RemoveVolume(podId string, record []byte) error {
 	return nil
 }
 
+type RawBlockStorage struct {
+	rootPath string
+}
+
+func RawBlockFactory(_ *dockertypes.Info, _ *daemondb.DaemonDB) (Storage, error) {
+	driver := &RawBlockStorage{
+		rootPath: filepath.Join(utils.HYPER_ROOT, "rawblock"),
+	}
+	return driver, nil
+}
+
+func (s *RawBlockStorage) Type() string {
+	return "rawblock"
+}
+
+func (s *RawBlockStorage) RootPath() string {
+	return s.rootPath
+}
+
+func (s *RawBlockStorage) Init() error {
+	if err := os.MkdirAll(filepath.Join(s.RootPath(), "volumes"), 0700); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (*RawBlockStorage) CleanUp() error { return nil }
+
+func (s *RawBlockStorage) PrepareContainer(containerId, sharedDir string) (*runv.VolumeDescription, error) {
+	devFullName := filepath.Join(s.RootPath(), "blocks", containerId)
+
+	vol := &runv.VolumeDescription{
+		Name:   devFullName,
+		Source: devFullName,
+		Fstype: "xfs",
+		Format: "raw",
+	}
+
+	return vol, nil
+}
+
+func (s *RawBlockStorage) CleanupContainer(id, sharedDir string) error {
+	return nil
+}
+
+func (s *RawBlockStorage) InjectFile(src io.Reader, mountId, target, baseDir string, perm, uid, gid int) error {
+	return fmt.Errorf("unsupported")
```

Comment:
The hyperd inject resolv.conf file with `InjectFile` by default. And we have inject file cases in CI scripts. I think we need this to be implemented finally.